code/modules/mob/status_procs.dm

```diff
@@ -187,29 +187,38 @@
 /////////////////////////////////// EYE_BLURRY ////////////////////////////////////
 
 /mob/proc/blur_eyes(amount)
+	var/need_update = (eye_blurry == 0 && amount > 0)
 	if(amount>0)
-		var/old_eye_blurry = eye_blurry
 		eye_blurry = max(amount, eye_blurry)
-		if(!old_eye_blurry)
-			overlay_fullscreen("blurry", /obj/screen/fullscreen/blurry)
+	if(need_update)
+		update_eye_blur()
 
 /mob/proc/adjust_blurriness(amount)
-	var/old_eye_blurry = eye_blurry
+	var/need_update = ((eye_blurry == 0 && amount > 0) || eye_blurry + amount <= 0)
 	eye_blurry = max(eye_blurry+amount, 0)
-	if(amount>0)
-		if(!old_eye_blurry)
-			overlay_fullscreen("blurry", /obj/screen/fullscreen/blurry)
-	else if(old_eye_blurry && !eye_blurry)
-		clear_fullscreen("blurry")
+	if(need_update)
+		update_eye_blur()
 
 /mob/proc/set_blurriness(amount)
-	var/old_eye_blurry = eye_blurry
+	var/need_update = ((eye_blurry == 0 && amount > 0) || (eye_blurry > 0 && amount <= 0))
 	eye_blurry = max(amount, 0)
-	if(amount>0)
-		if(!old_eye_blurry)
-			overlay_fullscreen("blurry", /obj/screen/fullscreen/blurry)
-	else if(old_eye_blurry)
-		clear_fullscreen("blurry")
+	if(need_update)
+		update_eye_blur()
+
+/mob/proc/update_eye_blur()
+	if(!client)
+		return
+	var/obj/screen/plane_master/open_turf/OT = locate(/obj/screen/plane_master/open_turf) in client.screen
+	var/obj/screen/plane_master/game_world/GW = locate(/obj/screen/plane_master/game_world) in client.screen
+	GW.filters = null
+	OT.filters = null
+	GW.backdrop(src)
+	OT.backdrop(src)
+	if(eye_blurry)
+		GW.filters += GAUSSIAN_BLUR
```

Comment:
Wouldn't it be more performant, since filter() is a proc, to just initialize them all into a global table at roundstart?